### ISSUE TITLE
is_constant() raises AssertionError for constant Poly instances

### ISSUE THREAD

**Issue by user234683** [sympy/sympy#12913]
```python
>>> Poly(3,x).is_constant()
Traceback (most recent call last):
  File "<pyshell#22>", line 1, in <module>
    Poly(3,x).is_constant()
  File "C:\Python34-64\lib\site-packages\sympy-1.1.1.dev0-py3.4.egg\sympy\core\expr.py", line 533, in is_constant
    assert not free
AssertionError
```

**Comment by parsoyaarihant** [comment]
You need to use sympy number instead of `int` see [here](http://docs.sympy.org/latest/gotchas.html#python-numbers-vs-sympy-numbers).
Correct way to do it would be:
```python
>>> Poly(S(3), x).as_expr().is_constant()
True
```

**Comment by smichr** [comment]
This is a consequence of #12846 and is fixed in #12881 but if (after that is committed) you want to add a test to confirm that this is closed, that would be good.

```
>>> Poly(3, x).free_symbols
{x}  # should be {}
```

**Comment by smichr** [comment]
A test for this can be added.
